File: app/src/main/java/br/edu/ifce/engcomp/francis/radarpolitico/models/Proposicao.java
package br.edu.ifce.engcomp.francis.radarpolitico.models;

/**
 * Created by francisco on 14/03/16.
 */
public class Proposicao {
    private String id;
    private String nome;

    private String sigla;
    private String numero;
    private String ano;

    private String tipoProposicao;
    private String tema;
    private String ementa;
    private String explicacaoEmenta;
    private String indexacao;

    private String regimeTramitacao;
    private String situacao;

    private String urlInteiroTeor;

    private String quantidadeAutores;

    private String idAutor;
    private String nomeAutor;

    private Votacao ultimaVotacao;

    public Proposicao() {

    }

    public void merge(Proposicao p){
        this.nome = p.nome == null? this.nome : p.nome;
        this.sigla = p.sigla == null? this.sigla : p.sigla;
        this.ano = p.ano == null? this.ano : p.ano;
        this.tipoProposicao = p.tipoProposicao == null? this.tipoProposicao : p.tipoProposicao;
        this.tema = p.tema == null? this.tema : p.tema;
        this.ementa = p.ementa == null? this.ementa : p.ementa;
        this.explicacaoEmenta = p.explicacaoEmenta == null? this.explicacaoEmenta : p.explicacaoEmenta;
        this.indexacao = p.indexacao == null? this.indexacao : p.indexacao;
        this.regimeTramitacao = p.regimeTramitacao == null? this.regimeTramitacao : p.regimeTramitacao;
        this.situacao = p.situacao == null? this.situacao : p.situacao;
        this.urlInteiroTeor = p.urlInteiroTeor == null? this.urlInteiroTeor : p.urlInteiroTeor;
        this.quantidadeAutores = p.quantidadeAutores == null? this.quantidadeAutores : p.quantidadeAutores;
        this.idAutor = p.idAutor == null? this.idAutor : p.idAutor;
        this.nomeAutor = p.nomeAutor == null? this.nomeAutor : p.nomeAutor;
        this.ultimaVotacao = p.ultimaVotacao == null? this.ultimaVotacao : p.ultimaVotacao;
    }

    public Votacao getUltimaVotacao() {
        return ultimaVotacao;
    }

    public void setUltimaVotacao(Votacao ultimaVotacao) {
        this.ultimaVotacao = ultimaVotacao;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getSigla() {
        return sigla;
    }

    public void setSigla(String sigla) {
        this.sigla = sigla;
    }

    public String getNumero() {
        return numero;
    }

    public void setNumero(String numero) {
        this.numero = numero;
    }

    public String getAno() {
        return ano;
    }

    public void setAno(String ano) {
        this.ano = ano;
    }

    public String getTipoProposicao() {
        return tipoProposicao;
    }

    public void setTipoProposicao(String tipoProposicao) {
        this.tipoProposicao = tipoProposicao;
    }

    public String getTema() {
        return tema;
    }

    public void setTema(String tema) {
        this.tema = tema;
    }

    public String getEmenta() {
        return ementa;
    }

    public void setEmenta(String ementa) {
        this.ementa = ementa;
    }

    public String getExplicacaoEmenta() {
        return explicacaoEmenta;
    }

    public void setExplicacaoEmenta(String explicacaoEmenta) {
        this.explicacaoEmenta = explicacaoEmenta;
    }

    public String getIndexacao() {
        return indexacao;
    }

    public void setIndexacao(String indexacao) {
        this.indexacao = indexacao;
    }

    public String getRegimeTramitacao() {
        return regimeTramitacao;
    }

    public void setRegimeTramitacao(String regimeTramitacao) {
        this.regimeTramitacao = regimeTramitacao;
    }

    public String getSituacao() {
        return situacao;
    }

    public void setSituacao(String situacao) {
        this.situacao = situacao;
    }

    public String getUrlInteiroTeor() {
        return urlInteiroTeor;
    }

    public void setUrlInteiroTeor(String urlInteiroTeor) {
        this.urlInteiroTeor = urlInteiroTeor;
    }

    public String getQuantidadeAutores() {
        return quantidadeAutores;
    }

    public void setQuantidadeAutores(String quantidadeAutores) {
        this.quantidadeAutores = quantidadeAutores;
    }

    public String getIdAutor() {
        return idAutor;
    }

    public void setIdAutor(String idAutor) {
        this.idAutor = idAutor;
    }

    public String getNomeAutor() {
        return nomeAutor;
    }

    public void setNomeAutor(String nomeAutor) {
        this.nomeAutor = nomeAutor;
    }

    @Override
    public String toString() {
        return "Proposicao{" +
                "id='" + id + '\'' +
                ", nome='" + nome + '\'' +
                ", sigla='" + sigla + '\'' +
                ", numero='" + numero + '\'' +
                ", ano='" + ano + '\'' +
                ", tipoProposicao='" + tipoProposicao + '\'' +
                ", tema='" + tema + '\'' +
                ", ementa='" + ementa + '\'' +
                ", explicacaoEmenta='" + explicacaoEmenta + '\'' +
                ", indexacao='" + indexacao + '\'' +
                ", regimeTramitacao='" + regimeTramitacao + '\'' +
                ", situacao='" + situacao + '\'' +
                ", urlInteiroTeor='" + urlInteiroTeor + '\'' +
                ", quantidadeAutores='" + quantidadeAutores + '\'' +
                ", idAutor='" + idAutor + '\'' +
                ", nomeAutor='" + nomeAutor + '\'' +
                ", ultimaVotacao=" + ultimaVotacao +
                '}';
    }
}

File: app/src/main/kotlin/br/edu/ifce/engcomp/francis/radarpolitico/models/Proposicao.kt
package br.edu.ifce.engcomp.francis.radarpolitico.models

import java.util.*
import android.os.Parcel
import android.os.Parcelable

import java.io.Serializable

/**
 * Created by francisco on 14/03/16.
 */
data class Proposicao (
        var id: String = "",
        var nome: String? = "",

        var sigla: String? = "",
        var numero: String? = "",
        var ano: String? = "",

        var tipoProposicao: String? = "",
        var tema: String? = "",
        var ementa: String? = "",
        var explicacaoEmenta: String? = "",
        var indexacao: String? = "",

        var regimeTramitacao: String? = "",
        var situacao: String? = "",

        var urlInteiroTeor: String? = "",
        var quantidadeAutores: String? = "",

        var idAutor: String? = "",
        var nomeAutor: String? = "",

        var dataVotacao: Date? = null,
        var ultimaVotacao: Votacao? = null) : Parcelable {

    constructor(source: Parcel): this(
            source.readString(),
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as Date?,
            source.readParcelable<Votacao?>(Votacao::class.java.classLoader))

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        dest?.writeString(id)
        dest?.writeSerializable(nome)
        dest?.writeSerializable(sigla)
        dest?.writeSerializable(numero)
        dest?.writeSerializable(ano)
        dest?.writeSerializable(tipoProposicao)
        dest?.writeSerializable(tema)
        dest?.writeSerializable(ementa)
        dest?.writeSerializable(explicacaoEmenta)
        dest?.writeSerializable(indexacao)
        dest?.writeSerializable(regimeTramitacao)
        dest?.writeSerializable(situacao)
        dest?.writeSerializable(urlInteiroTeor)
        dest?.writeSerializable(quantidadeAutores)
        dest?.writeSerializable(idAutor)
        dest?.writeSerializable(nomeAutor)
        dest?.writeSerializable(dataVotacao)
        dest?.writeParcelable(ultimaVotacao, 0)
    }

    companion object {
        @JvmField final val CREATOR: Parcelable.Creator<Proposicao> = object : Parcelable.Creator<Proposicao> {
            override fun createFromParcel(source: Parcel): Proposicao {
                return Proposicao(source)
            }

            override fun newArray(size: Int): Array<Proposicao?> {
                return arrayOfNulls(size)
            }
        }
    }

    fun merge(p: Proposicao) {
        this.nome = if (p.nome == null) this.nome else p.nome;
        this.numero = if (p.numero == null) this.numero else p.numero;
        this.sigla = if (p.sigla == null) this.sigla else p.sigla;
        this.ano = if (p.ano == null) this.ano else p.ano;
        this.tipoProposicao = if (p.tipoProposicao == null) this.tipoProposicao else p.tipoProposicao;
        this.tema = if (p.tema == null) this.tema else p.tema;
        this.ementa = if (p.ementa == null) this.ementa else p.ementa;
        this.explicacaoEmenta = if (p.explicacaoEmenta == null) this.explicacaoEmenta else p.explicacaoEmenta;
        this.indexacao = if (p.indexacao == null) this.indexacao else p.indexacao;
        this.regimeTramitacao = if (p.regimeTramitacao == null) this.regimeTramitacao else p.regimeTramitacao;
        this.situacao = if (p.situacao == null) this.situacao else p.situacao;
        this.urlInteiroTeor = if (p.urlInteiroTeor == null) this.urlInteiroTeor else p.urlInteiroTeor;
        this.quantidadeAutores = if (p.quantidadeAutores == null) this.quantidadeAutores else p.quantidadeAutores;
        this.idAutor = if (p.idAutor == null) this.idAutor else p.idAutor;
        this.nomeAutor = if (p.nomeAutor == null) this.nomeAutor else p.nomeAutor;
        this.ultimaVotacao = if (p.ultimaVotacao == null) this.ultimaVotacao else p.ultimaVotacao;
    }
}
